documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -52,178 +48,167 @@ public void run(String routingKey, String message) {
+
+- Scope Name.
+- Stream Name.
+- Stream Configuration.
+
+The most interesting task is to create the Stream Configuration (`streamConfig`). Like many objects in Pravega, a Stream takes a configuration object that allows
 a developer to control various behaviors of the Stream.  All configuration
-objects in Pravega use a builder pattern for construction.  There are really two
-important configuration items related to streams: Retention Policy and Scaling
-Policy.  
-
-Retention Policy allows the developer to control how long data is kept in a
-Stream before it is deleted.  S/he can specify data should be kept for a certain
-period of time (ideal for situations like regulatory compliance that mandate
-certain retention periods) or to retain data until a certain number of bytes
-have been consumed.  At the moment, Retention Policy is not completely
-implemented.  By default, the RetentionPolicy is set as "unlimited" meaning, data
-will not be removed from the Stream.
-
-Scaling Policy is the way developers configure a Stream to take advantage
-Pravega's auto-scaling feature.  In line 6, we use a fixed policy, meaning the
-Stream is configured with the given number of Stream Segments and that won't
-change.  The other options are to scale by a given number of Events per second
-or a given number of Kilobytes per second.  In these two policies, the developer
-specifies a target rate, a scaling factor and a minimum number of Segments.  The
-target rate is straight forward, if ingest rate exceeds a certain number of
-Events or Kilobytes of data for a sustained period of time, Pravega will attempt
-to add new Stream Segments to the Stream.  If the rate drops below that
-threshold for a sustained period of time, Pravega will attempt to merge adjacent
-Stream Segments.  The scaling factor is a setting on the Scaling Policy that
-determines how many Stream Segments should be added when the target rate (of
-Events or Kilobytes) is exceeded.  The minimum number of Segments is a factor
-that sets the minimum degree of read parallelism to be maintained; if this value
-is set at 3, for example, there will always be 3 Stream Segments available on
-the Stream.  Currently, this property is effective only when the stream is
+objects in Pravega use a builder pattern for construction. **Retention Policy** and **Scaling
+Policy** are the two important configuration items related to streams.   
+
+**Retention Policy:**  Pravega supports the following two types of Retentions:
+
+ - **Time based Retention**: It allows the developer to control how long the data is kept in a Stream before it is deleted. The developer can specify the time limit (milliseconds) to keep the data for a certain period of time (ideal for situations like regulatory compliance that mandate
```

Comment:
Stream  -> `Stream` ?

Reply:
@RaulGracia What i should edit here

Reply:
The format of the word Stream to be `Stream`.

Reply:
Sorry, I got confused with the format convention. From my viewpoint, the change needed here is the same as in the case of Scaling policies: If these concepts are explained in depth in `Concepts.md`, we don't need to duplicate the text here. Thus, we may only need to provide a couple of sentences about the concept (what is a Retention or Scaling policy at a high level) and then refer the user to the `Concepts.md` page via a link.

Reply:
For scaling, i have provided link.
Let the retention content  remain here(As retention is not discussed in detail in Pravega Concepts).

Reply:
I propose to move this text about retention policies to the last part of `Concepts.md` (`A Note on Tiered Storage`). Something like:

```
Pravega allows users to store data in Tier 2 as long as there is storage capacity available. But sometimes, users may not be interested to keep all the historical data related to a Stream. Instead, there are use-cases in which it may be useful to retain just a fraction of a Stream's data. For this reason, Streams can be configured with **Retention Policies**.  

Pravega supports two types of Retention Policies:

 - **Time-based Retention**: It allows the developer to control for how long the data is kept in a Stream before it is deleted. The developer can specify the time limit (milliseconds) in the Stream policy, which is ideal for situations like regulatory compliance that mandate certain retention periods.

- **Size-based Retention**: Retains the newest subset of a Stream's data that does not exceed the specified size in bytes.
```

After adding that at the end of `Concepts.md`, here you can just say that Pravega supports both Scaling and Retention policies, with the corresponding links.

Reply:
Accepted suggestion